test/cluster/wait_for_skupper_service_test.go

```diff
@@ -1,114 +0,0 @@
-package cluster
-
-import (
-	"fmt"
-	"testing"
-	"time"
-
-	"gotest.tools/assert"
-	apiv1 "k8s.io/api/core/v1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/wait"
-)
-
-type ClusterContextMock struct {
-	calledWithService string
-	calledWithRetryFn func() (*apiv1.Service, error)
-}
-
-const (
-	expectedError string = "some error"
-)
-
-var (
-	mockSucceeds bool
-	mockService  *apiv1.Service = &apiv1.Service{
-		ObjectMeta: v1.ObjectMeta{
-			//Name: "mockName",
-		},
-	}
-	defaultDuration = 10 * time.Millisecond
-)
-
-func (cc *ClusterContextMock) waitForSkupperServiceToBeCreated(name string, retryFn func() (*apiv1.Service, error), backoff wait.Backoff) (*apiv1.Service, error) {
-	cc.calledWithService = name
-	cc.calledWithRetryFn = retryFn
-
-	mockService.Name = name
-	if mockSucceeds {
-		return mockService, nil
-	}
-	return nil, fmt.Errorf(expectedError)
-
-}
-
-func TestWaitForSkupperServiceToBeCreatedAndReadyToUse(t *testing.T) {
```

Comment:
as mentioned in other comments, I would keep this test, and this original function until we find a better solution...